sotodlib/io/g3tsmurf_utils.py

```diff
@@ -129,13 +138,41 @@ def get_batch(
     """
     
     session = archive.Session()
-    obs = session.query(Observations).filter(Observations.obs_id==obs_id).one()
+
+    if isinstance(obs_id, Observations):
+        obs = obs_id
+        obs_id = obs.obs_id
+    else:
+        obs = session.query(Observations).filter(Observations.obs_id==obs_id).one()
     db_files = session.query(Files).filter(Files.obs_id==obs_id).order_by(Files.start)
     filenames = sorted( [f.name for f in db_files])
     
     ts = obs.tunesets[0] ## if this throws an error we have some fallbacks
     obs_dets, obs_samps = len(ts.dets), obs.n_samples
     session.close()
+
+    ## should we let folks overwrite this here?
+    if "archive" in load_file_args:
+        archive = load_file_args.pop("archive")
+        
+    if "status" in load_file_args:
+        status = load_file_args.pop("status")
+    else:
+        status = SmurfStatus.from_file(filenames[0])
+
+    if start == None:
+        start = obs.start
```

Comment:
`if start is None:` (I don't think == is standard for getting Nones in python)

if start is None can we just set `samp_s` to 0 and skip the math? that will avoid rounding errors and missed edge cases by default. Same `if stop is None`, just set `samp_e` to `obs_samps`.